### DISCUSSION TITLE
[@types/lodash] Lodash set Issue

### DISCUSSION THREAD

**Discussion by RaffB2** [DefinitelyTyped/DefinitelyTyped#62251] [Issues with a @types package]
Hey, we're using Lodash/fp and we're running into an issue with the set function, if we assign something new or replace something in an object, the type doesn't match the expected outcome. Therefore, validation is impossible.
Example:
```
const obj = { foo: 1 };
const newObj = set('bar')(2)(obj);
```
The type assigned to newObj is 
```
const newObj: {
    foo: number;
}
```
We've also tried setting the object type manually like this:
```
interface NewObjType {
  foo:number,
  bar:number
}
const obj = { foo: 1 };
const newObj = set('bar')(2)<NewObjType>(obj);
```
Which assigns the correct type to newObj, but we're not sure if the path or value matches the expected type.
Anyone knows a solution to this?

[a comment by an automated account is not shown]

**Comment by aj-r** [comment]
Unfortunately it's very difficult (maybe impossible?) to get the `fp` types correct in cases like this. For now the explicit generic type is the only solution I'm aware of. If anyone has a clever idea for how to fix the types (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/fp.d.ts#L154) please send a PR.
